src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/UserGroupInformationWrapper.scala

```diff
@@ -0,0 +1,30 @@
+
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.api.rest.services
+
+import org.apache.hadoop.security.UserGroupInformation
+import org.springframework.stereotype.Service
+
+trait UserGroupInformationWrapper {
+  def loginUserFromKeytab(principal: String, keytab: String): Unit
+}
+
+@Service
+class UserGroupInformationWrapperImpl extends UserGroupInformationWrapper {
+  override def loginUserFromKeytab(principal: String, keytab: String): Unit =
+    UserGroupInformation.loginUserFromKeytab(principal, keytab)
```

Comment:
Every time it tries to read, it logins and refreshes ticket. 
That is not necessary, you can check if ticket is valid if yes just continue